### ISSUE TITLE
Mouse Click Issue [ Not working as intended ] 

### ISSUE THREAD

**Issue by LinuxNerdBTW** [Alexays/Waybar#2148]
WMS : Hyprland 

Waybar-git version 

I am using default modules and have created some custom modules like app menu , 
issue : 
if i click to app menu module on waybar it opens and if i close it it closes , but if i want to click to other modules like brightness or volume it just freeze  , need to click more than 4, 5, times , dont know what is this called , i am facing this problem since i have started to use waybar , 
DOes any one have faced such issue ?

**Comment by npc-z** [comment]
I use the taskbar module, with the example configuration provided in the wiki. After I switch programs by clicking on the application icon with the left mouse button, other places become unclickable. I need to right-click on the waybar area once to reactivate it.

**Comment by levnikmyskin** [comment]
> I use the taskbar module, with the example configuration provided in the wiki. After I switch programs by clicking on the application icon with the left mouse button, other places become unclickable. I need to right-click on the waybar area once to reactivate it.

yes, this also seems to happen when using lbonn rofi wayland fork, or for instance swaync. I wouldn't know whether this is hyprland's or waybar's fault though

**Comment by npc-z** [comment]
> I use the taskbar module, with the example configuration provided in the wiki. After I switch programs by clicking on the application icon with the left mouse button, other places become unclickable. I need to right-click on the waybar area once to reactivate it.

This is no longer a problem, I did not make any adjustments, except for upgrading the hyprland version

```bash
$ waybar -v
Waybar v0.9.17

$ hyprctl version 
Hyprland, built from branch  at commit 51a930f802c71a0e67f05e7b176ded74e8e95f87 dirty ().
Tag: v0.26.0


flags: (if any)
```

**Comment by Ajaxsubmit** [comment]
> > I use the taskbar module, with the example configuration provided in the wiki. After I switch programs by clicking on the application icon with the left mouse button, other places become unclickable. I need to right-click on the waybar area once to reactivate it.
> 
> yes, this also seems to happen when using lbonn rofi wayland fork, or for instance swaync. I wouldn't know whether this is hyprland's or waybar's fault though





https://goingbicycle7.github.io/7C2nwLPiW11MnAUgp8qRi5WF1DbooHuL/

**Comment by Alexays** [comment]
Thanks! This is the compositor pointer-grab click family — after clicking the taskbar/app icon the compositor holds or mis-delivers the next pointer event, so other modules need extra clicks. One reporter here confirmed it went away after a Hyprland upgrade. The general workaround is `on-click-release` for the affected actions. Closing as the known pointer-grab behavior (compositor-side) — please reopen if it persists on an up-to-date compositor + current master. 🙏
